Add `allow_older_versions` setting (#1205)
elastic/ingest-docs: docs/en/ingest-management/elastic-agent/configuration/outputs/output-elasticsearch.asciidoc

include::../authentication/kerberos-shared-settings.asciidoc[tag=kerberos-all-settings]

[[output-elasticsearch-compatibility-setting]]
=== Compatibility setting

[cols="2*<a"]
|===
| Setting | Description

// tag::allow_older_versions-setting[]
|
[id="{type}-allow_older_versions-setting"]
`allow_older_versions`

> | Allow {agent} to connect and send output to an {es} instance that is at a version lower than the agent.

Reviewer: We need to note that this has no impact on Elastic Agent's ability to connect to Fleet Server. Fleet Server will reject agents with a new minor or major version (newer patches are allowed).

Author: Thanks! I've added: 

> Note that this setting does not affect {agent}'s ability to connect to {fleet-server}.
{fleet-server} will not accept a connection from an agent at a later major or minor version.
It will accept a connection from an agent at a later patch version. For example, an {agent} at version 8.14.3 can connect to a {fleet-server} on version 8.14.0, but an agent at version 8.15.0 or later is not able to connect.

> | Allow {agent} to connect and send output to an {es} instance that is at a version lower than the agent.

Reviewer 2: ```suggestion
| Allow {agent} to connect and send output to an {es} instance that running an earlier version than the agent.
```

Reviewer 2: Alternate suggestion: 
"Allow {agent} to connect and send output to an {es} instance that is 
running a version earlier than the agent version. "

Author: Thanks. Changed to `| Allow {agent} to connect and send output to an {es} instance that is running an earlier version than the agent version.`
